greenery/fsm_test.py

```diff
@@ -583,7 +619,13 @@ def test_dead_default():
     blockquote | blockquote
     blockquote & blockquote
     blockquote ^ blockquote
-    reversed(blockquote)
+    # Fsm does not support the `Reversible` protocol, because its
+    # `__reversed__` implementation does not return an iterator.
```

Comment:
It looks like `Fsm` is misusing `__reversed__`. Since `Fsm` isn't part of `greenery`'s public API at present we could probably change this fairly painlessly. For `Pattern`, we can't do this without a breaking change but maybe that'll happen sometime later.